runelite-client/src/main/java/net/runelite/client/plugins/worldhopper/WorldSwitcherPanel.java

```diff
@@ -214,12 +217,33 @@ void resetAllFavoriteMenus()
 
 	void populate(List<World> worlds)
 	{
+		if (!rows.isEmpty())
+		{
+			pingHistory.clear();
+
+			for (WorldTableRow row : rows)
+			{
+				pingHistory.put(row.getWorld(), row.getPing());
+			}
+		}
+
 		rows.clear();
 
 		for (int i = 0; i < worlds.size(); i++)
 		{
 			World world = worlds.get(i);
-			rows.add(buildRow(world, i % 2 == 0, world.getId() == plugin.getCurrentWorld() && plugin.getLastWorld() != 0, plugin.isFavorite(world)));
+			Integer ping;
+
+			if (!pingHistory.containsKey(world))
+			{
+				ping = 0;

```

Comment:
just initialize ping to 0

Reply:
Went with .getOrDefault() 
Seemed to be the same result with less code.